tests/c_api_tests/test_cpp_api_meta_data.cpp

```diff
@@ -0,0 +1,80 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#define PGM_ENABLE_EXPERIMENTAL
+
+#include "power_grid_model_cpp.hpp"
+
+#include <doctest/doctest.h>
+
+#include <memory>
+
+namespace power_grid_model_cpp {
+
+TEST_CASE("C++ API Meta Data") {
+    SUBCASE("Datasets") {
+        // check dataset
+        for (Idx idx_dataset = 0; idx_dataset != MetaData::n_datasets(); ++idx_dataset) {
+            MetaDataset const* const dataset = MetaData::get_dataset_by_idx(idx_dataset);
+            std::string const dataset_name = MetaData::dataset_name(dataset);
+            CHECK(MetaData::get_dataset_by_name(dataset_name) == dataset);
+            CAPTURE(dataset_name);
+            // manually check a few to remove core dependency
```

Comment:
```suggestion
            // check a few dataset types
```
this type of comment should not be done inline. I personally do this in a github comment on the PR for reviewers to know